pyHS100/pyHS100.py

```diff
@@ -480,3 +552,280 @@ def state_information(self):
+        """
+
+        return self.sysinfo["deviceId"]
+
+    @device_id.setter
+    def device_id(self, deviceid):
+        """
+        Sets the device id.
+
+        :param str deviceid: Device id
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper(
+            "system",
+            "set_device_id",
+            {"deviceId": deviceid}
+        )
+
+    @property
+    def hardware_id(self):
+        """
+        Gets the device hardware id.
+
+        :return: Hardware id
+        :rtype: str
+        :raises SmartPlugException: on error
+        """
+
+        return self.sysinfo["hwId"]
+
+    @hardware_id.setter
+    def hardware_id(self, hwid):
+        """2
+        Sets the device id.
+
+        :param str hwid: Device hardware id
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "set_hw_id", {"hwId": hwid})
+
+    def test_uboot(self):
+        """
+        Perform uBoot boot loader check.
+
+        :return: None
+        :rtype: None
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "test_check_uboot")
+
+    def firmware_download_url(self, url):
+        """
+        Download Firmware from URL
+
+        :param str url: URL for firmware to download
+        :return: None
+        :rtype: None
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "download_firmware", {"url": url})
+
+    def firmware_download_flash(self):
+        """
+        Flashes the downloaded firmware.
+
+        :return: None
+        :rtype: None
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "flash_firmware", {})
+
+    def firmware_download_state(self):
+        """
+        Gets the firmware download state.
+
+        :return: Download state
+        :rtype: str
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "get_download_state", {})
+
+    def check_config(self):
+        """
+        Checks devices config.
+
+        :return: None
+        :rtype: None
+        :raises SmartPlugException: on error
+        """
+
+        self._query_helper("system", "check_new_config")
+
+    def scan_ap(self, refresh=1):
+        """
+        Scans for access points.
+
+        :param int refresh: Refresh interval in seconds.
+        :return: Returns a list of access points.
+        :rtype: list
+        :raises SmartPlugException: on error
+        """
+
+        return self._query_helper(
+            "netif",
+            "get_scaninfo",
+            {"refresh": refresh}
+        )
+
+    # TODO: map out the key_types
+    def connect_ap(self, ssid='', password='', key_type=0):
+        """
+        Connects to an access point.
+
+        :param str ssid: Access point ssid.
+        :param str password: Access point password.
+        :param int key_type: Access point encryption type.
```

Comment:
I do not know what the acceptable numbers are. there is no documentation on this. this would be one of the many reasons why i would need an actual device to test. there is no existing documentation of what numbers are what security type